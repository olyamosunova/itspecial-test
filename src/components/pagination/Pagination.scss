@import "./src/styles/_variables.scss";

.pagination {
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
  align-items: stretch;
  margin: 15px 0;
  padding: 0;

  list-style: none;
}

.pagination__item {
  &:first-of-type .pagination__link {
    border-top-left-radius: 0.25rem;
    border-bottom-left-radius: 0.25rem;
  }

  &:last-of-type .pagination__link {
    border-top-right-radius: 0.25rem;
    border-bottom-right-radius: 0.25rem;
  }

  &:not(:first-of-type) .pagination__link {
    margin-left: -1px;
  }

  &--active .pagination__link {
    color: $button-text-color;
    background-color: $button-background;
    border-color: $button-background;
  }
}

.pagination__link {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 0.375rem 0.75rem;

  font-size: 0.95rem;
  color: $button-background;
  text-decoration: none;

  background-color: $button-text-color;
  border: 1px solid $border-color;

  outline: none;

  &:disabled {
    opacity: 0.5;
  }

  &:not(:disabled) {
    cursor: pointer;
  }
}
