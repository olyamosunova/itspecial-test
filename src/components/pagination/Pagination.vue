<template>
    <nav class="d-flex justify-content-end">
        <ul class="pagination">
            <li class="page-item">
                <button class="page-link" type="button" aria-label="Previous" @click="handlerClickChangePage('prev')" :disabled="currentPage === 1">
                    <span aria-hidden="true">&laquo;</span>
                </button>
            </li>
            <li class="page-item" v-bind:class="{'active': currentPage === page}" v-for="page in pages" :key="page">
                <button class="page-link" type="button" @click="handlerClickChangePage('', page)">{{page}}</button>
            </li>
            <li class="page-item">
                <button class="page-link" type="button" aria-label="Next" @click="handlerClickChangePage('next')" :disabled="currentPage === pages.length">
                    <span aria-hidden="true">&raquo;</span>
                </button>
            </li>
        </ul>
    </nav>
</template>

<script>
    export default {
        name: "Pagination",
        props: {
            pages: Array
        },
        data() {
            return {}
        },
        computed: {
            currentPage() {
                return this.$store.getters.currentPage;
            }
        },
        methods: {
            handlerClickChangePage(type, page) {
                this.$emit('onClickChangePage', {type, page})
            }
        }
    }
</script>

<style lang="scss">
    .page-item {
        button:disabled:hover {
            background-color: inherit;
            border: 1px solid #dee2e6;
        }
    }
</style>
