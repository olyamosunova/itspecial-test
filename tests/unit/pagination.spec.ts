import { mount, createLocalVue } from "@vue/test-utils";
import Vuex from "vuex";
import Pagination from "@/components/pagination/Pagination.vue";

const localVue = createLocalVue();
localVue.use(Vuex);

const pages = [1, 2, 3];

describe('unit tests for Pagination component', () => {
    let store: any;
    let getters;

    beforeEach(() => {
        getters = {
            currentPage: () => 1
        };

        store = new Vuex.Store({ getters });
    });

    it('should match the snapshot', () => {
       const wrapper = mount(Pagination, { store, localVue, propsData: { pages } });
       expect(wrapper.html()).toMatchSnapshot();
    });
});
