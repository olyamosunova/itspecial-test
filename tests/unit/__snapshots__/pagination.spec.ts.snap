// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit tests for Pagination component should match the snapshot 1`] = `
<nav class="d-flex justify-content-end">
  <ul class="pagination">
    <li class="page-item"><button type="button" aria-label="First" disabled="disabled" class="page-link"><span aria-hidden="true">«</span></button></li>
    <li class="page-item"><button type="button" aria-label="Previous" disabled="disabled" class="page-link"><span aria-hidden="true">‹</span></button></li>
    <!---->
    <li class="page-item active"><button type="button" class="page-link">
        1</button></li>
    <li class="page-item"><button type="button" class="page-link">
        2</button></li>
    <li class="page-item"><button type="button" class="page-link">
        3</button></li>
    <!---->
    <li class="page-item"><button type="button" aria-label="Next" class="page-link"><span aria-hidden="true">›</span></button></li>
    <li class="page-item"><button type="button" aria-label="Last" class="page-link"><span aria-hidden="true">»</span></button></li>
  </ul>
</nav>
`;
